GPU/Common/FramebufferManagerCommon.cpp

```diff
@@ -551,20 +551,33 @@ void FramebufferManagerCommon::CopyToDepthFromOverlappingFramebuffers(VirtualFra
 	// For now, let's just do the last thing, if there are multiple.
 
 	// for (auto &source : sources) {
-	if (sources.size()) {
+	if (!sources.empty()) {
+		draw_->InvalidateCachedState();
+
 		auto &source = sources.back();
 		if (source.channel == RASTER_DEPTH) {
 			// Good old depth->depth copy.
 			BlitFramebufferDepth(source.vfb, dest);
 			gpuStats.numDepthCopies++;
 			dest->last_frame_depth_updated = gpuStats.numFlips;
-		} else if (source.channel == RASTER_COLOR) {
+		} else if (source.channel == RASTER_COLOR && draw_->GetDeviceCaps().fragmentShaderDepthWriteSupported) {
 			VirtualFramebuffer *src = source.vfb;
-			// Copying color to depth.
 			if (src->drawnFormat != GE_FORMAT_565) {
 				WARN_LOG_ONCE(not565, G3D, "Drawn format of buffer at %08x not 565 as expected", src->fb_address);
 			}
-			BlitUsingRaster(src->fbo, 0.0f, 0.0f, src->renderWidth, src->renderHeight, dest->fbo, 0.0f, 0.0f, dest->renderWidth, dest->renderHeight, false, DRAW2D_565_TO_DEPTH, "565_to_depth");
+
+			// Really hate to do this, but tracking the depth swizzle state across multiple
+			// copies is not easy.
```

Comment:
I don't think there's any way the mirrors are slow, and ignorance would be strange since it uses them earlier in the frame IIRC.. 

But hm, if we always do the flip when converting depth to color when the bits aren't set... I'll work more on this tomorrow or Monday.